src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -16,6 +16,22 @@ public PackageSourceMappingProvider(ISettings settings)
             _settings = settings ?? throw new ArgumentNullException(nameof(settings));
         }
 
+        /// <summary>
+        /// Supports the disabling of saving to disk for any <see cref="ISettings"/> changes.
+        /// </summary>
+        /// <param name="shouldSkipSave">True to avoid saving any changes to disk and only modify the <see cref="ISettings"/> in memory.
+        /// Default is false.</param>
+        public PackageSourceMappingProvider(ISettings settings, bool shouldSkipSave)
+            : this(settings)
+        {
+            ShouldSkipSave = shouldSkipSave;
+        }
+
+        /// <summary>
+        /// Avoid saving to disk and only modify the <see cref="ISettings"/> in memory.
+        /// </summary>
+        public bool ShouldSkipSave { get; private set; }
```

Comment:
> This inconsistency is confusing, especially when SavePackageSourceMappings is called in places where there are comments that say it is specifically not writing to disk. 

I could only find one comment in this helper method, which is technically accurate, but I removed it based on your feedback as well as the fact that I'm not enforcing the helper to receive a provider which has `ShouldSkipSave==false`.
https://github.com/NuGet/NuGet.Client/blob/5e03c332991fcf02574f9e8b7a3eb7918aea91a5/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L3190

> Consider updating ShouldSkipSave to use a different prefix such as SkipPersisting or SkipWritingToDisk

The nomenclature "save" does have multiple meanings in our repo. We save changes to an `ISettings` in memory, and we also `SaveToDisk`. At least we're explicit about saving to disk, since we want that to be obvious. I don't think there's a common understanding that "save" must always mean "to disk". If so, we should probably update the terms across the entire repo at some point. 
The naming I've used here was chosen to be consistent with other providers, such as https://github.com/NuGet/NuGet.Client/blob/63edda646848553467639abe6bbade5ebc5dec61/src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs#L464